URI/ByteString/Extension.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module URI.ByteString.Extension where
+
+import Data.ByteString (ByteString)
+import Data.String (IsString(..))
+import Data.Text (Text)
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+import Lens.Micro
+
+import qualified Data.ByteString.Char8 as C8
+
+import URI.ByteString
+
+instance IsString Scheme where
+    fromString = Scheme . fromString
+
+instance IsString Host where
+    fromString = Host . fromString
+
+instance IsString (URIRef Absolute) where
+    fromString = either (error . show) id
+        . parseURI strictURIParserOptions
+        . C8.pack
+
+instance IsString (URIRef Relative) where
+    fromString = either (error . show) id
+        . parseRelativeRef strictURIParserOptions
+        . C8.pack
+
+fromText :: Text -> Maybe URI
+fromText = either (const Nothing) Just
```

Comment:
Is there really no function `Either a b -> Maybe b`? Could've sworn this existed.

Reply:
You can do it with Lens or the [either](https://www.stackage.org/haddock/lts-9.6/either-4.4.1.1/Data-Either-Combinators.html) package. I didn't feel it necessary to bring either (heh) of those in. Do you think I should bring in `either` for it?

Reply:
I don't think it's worth the dependency.

Reply:
Agree with @jferris